code/parse/sexp.cpp

```diff
@@ -19804,6 +19805,36 @@ void multi_sexp_set_respawns()
 	}
 }
 
+/**
+ * set a hotkey for one or more ships/wings
+ */
+void sexp_set_hotkey(int node)
+{
+	int objnum, setnum, n = node;
+	setnum = eval_num(n);
+	n   = CDR(n);
+
+	// first, look for ship name -- if found, then set hotkey to ship
+	// if not a ship, look for wing name, then set hotkey to each ship in wing
+	// note, hud_target_hotkey_add_remove() checks if the object has arrived or is destroyed or departed
+	for (; n != -1; n = CDR(n)) {
+		object_ship_wing_point_team oswpt;
+		sexp_get_object_ship_wing_point_team(&oswpt, CTEXT(n));
+		if (oswpt.type == OSWPT_TYPE_SHIP) {
+			objnum = oswpt.shipp->objnum;
+			if (objnum != -1)
+				hud_target_hotkey_add_remove(setnum, &Objects[objnum], HOTKEY_MISSION_FILE_ADDED);
+		}
+		if (oswpt.type == OSWPT_TYPE_WING) {
+			for (int i = 0; i < oswpt.wingp->current_count; i++) {
+				objnum = Ships[oswpt.wingp->ship_index[i]].objnum;
+				if (objnum != -1)
+					hud_target_hotkey_add_remove(setnum, &Objects[objnum], HOTKEY_MISSION_FILE_ADDED);
```

Comment:
Using `HOTKEY_MISSION_FILE_ADDED` (both here and line 19826) means that, if there are any user-added ships in the hotkey set, this SEXP will be ignored; is that desired behavior here?

Reply:
ATM that is the desired behavior. I was debating on whether to use that or `HOTKEY_USER_ADDED`., and I am very open to thoughts about it, because it's really a debate about who to favor, the mission designer or the player. My thinking was that if the mission designer specifies a hotkey and tells the player to press that key, then they want to ensure that the player didn't bind that key already, because then it might be confusing. In contrast though, the player might be annoyed that their hotkey just went away. Perhaps I should add a bool argument that is something along the lines of `override`, so if true then it uses `HOTKEY_MISSION_FILE_ADDED` and if false it uses `HOTKEY_USER_ADDED`?

Reply:
Oh wait.. I see what you mean. So my reasoning is still valid, but the to get the sexp to override player hotkeys (ie already existing ones) I would used `HOTKEY_USER_ADDED`.

Reply:
You could also add a new `#define` like, say, `HOTKEY_SEXP_ADDED` or something, and give it the exact behavior you want.